### PR TITLE
Feature/wallet sdk tests

### PR DESCRIPTION
**Pull request by jklein24**

### CODE REVIEW

File: packages/wallet-sdk/src/__tests__/helpers/wallet.helpers.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    if (walletStatus !== WalletStatus.DEPLOYED && walletStatus !== WalletStatus.READY) {
+        console.log(
+            `Not initializing because the wallet status is ${loginOutput.wallet.status}`
+        )
+    }
+
+    let content = `\n# Wallet for user ${userId}:\n# accountID: ${credentials.accountId}\n# test: ${test}\n`;
+
+    content += `export LIGHTSPARK_JWT_${userId}="${token}"\n`;
+    // process.env[`LIGHTSPARK_JWT_${userId}`] = token;
```
Comment (jklein24) [inline packages/wallet-sdk/src/__tests__/helpers/wallet.helpers.ts:120]: nit: remove commented-out code.

File: packages/wallet-sdk/tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "@lightsparkdev/tsconfig/base.json",
   "compilerOptions": {
+    "exactOptionalPropertyTypes": false,
```
Comment (jklein24) [inline packages/wallet-sdk/tsconfig.json:4]: Why do we need this?

File: packages/wallet-sdk/src/__tests__/helpers/wallet.helpers.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    )
+
+    let walletStatus= loginOutput.wallet.status
+
+    if (loginOutput.wallet.status === WalletStatus.NOT_SETUP) {
+        walletStatus = await client.deployWalletAndAwaitDeployed();
+    }
+
+    while (walletStatus === WalletStatus.INITIALIZING || walletStatus === WalletStatus.DEPLOYING) {
+        await sleep(WALLET_STATUS_TIMEOUT)
```
Comment (jklein24) [inline packages/wallet-sdk/src/__tests__/helpers/wallet.helpers.ts:102]: Something doesn't seem right here. The `deployWalletAndAwaitDeployed` and the `initializeWalletAndAwaitReady` functions automatically wait for the right state and time out if appropriate. I don't think you should need this sleeping block. Was something not working?

**Review by jklein24 — COMMENTED** [review]
Just a quick first glance, but the content mostly looks good!

File: packages/wallet-sdk/tsconfig.json (hunk 2 above)
Comment (coreymartin) [reply packages/wallet-sdk/tsconfig.json:4]: Yeah I'd prefer to leave this on unless it's a burden for some reason. We'd like to have unified config for our projects as much as possible and this rule is useful in some cases.

Comment (lukachi) [reply packages/wallet-sdk/tsconfig.json:4]: yeah, tests wouldn't run without overriding this property from base tsconfig

File: packages/wallet-sdk/src/__tests__/helpers/wallet.helpers.ts (hunk 3 above)
Comment (lukachi) [reply packages/wallet-sdk/src/__tests__/helpers/wallet.helpers.ts]: As i remember, `deployWalletAndAwaitDeployed` sometimes returns `DEPLOYING` status, but not `DEPLOYED`, i can remove this polling if it was fixed.

Comment (jklein24) [reply packages/wallet-sdk/src/__tests__/helpers/wallet.helpers.ts]: That should never be the case. If it is, there's a bug.

File: packages/wallet-sdk/src/__tests__/general_testnet.test.ts [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+let createDeployAndInitWalletResponse: Awaited<ReturnType<typeof createDeployAndInitWallet>> | undefined = undefined
+
+const ENCODED_REQUEST_FOR_TESTS = 'lnbcrt500n1pjdyx6tpp57xttmwwfvp3amu8xcr2lc8rs7zm26utku9qqh7llxwr6cf5yn4ssdqjd4kk6mtdypcxj7n6vycqzpgxqyz5vqsp5mdp46gsf4r3e6dmy7gt5ezakmjqac0mrwzunn7wqnekaj2wr9jls9qyyssq2cx3pzm3484x388crrp64m92wt6yyqtuues2aq9fve0ynx3ln5x4846agck90fnp5ws2mp8jy4qtm9xvszhcvzl7hzw5kd99s44kklgpq0egse'
+
+describe('Sanity tests', () => {
+
+    const invoiceData = {} as CreatedInvoiceData
+
+    jest.spyOn(authorizedClientWithLockedWallet, 'isAuthorized').mockResolvedValue(true)
```
[the remark this answers is not in the document]
Comment (lukachi) [reply packages/wallet-sdk/src/__tests__/general_testnet.test.ts]: we've separated authorized-unlocked, non authorized and authorized-locked clients to test validations of some client methods.

File: packages/wallet-sdk/tsconfig.json (hunk 2 above)
Comment (coreymartin) [reply packages/wallet-sdk/tsconfig.json]: Whatever the issue was, seems resolved now - I turned off the override and everything works.

**Review by coreymartin — APPROVED** [review]

**Pull request merged**